### ISSUE TITLE
Failed to read the dump file

### ISSUE THREAD

**Issue by jzatarain** [techtacoriginal/offlinebay#18]
I just downloaded the app for the first time, and when I opened it and searched for something, I only got an error saying: "Failed to read the dump file. Possible corruption or file doesn't exist."
What do I need to do to work this out?

**Comment by fckthisshieeeeet** [comment]
Yeah i got the same issue : i tried many way to get .csv dump files and i tried to update with the option in the software and i tried all of this with a vpn for torrent and each time it told me "Dump file invalid. Mismatching header" or Failed to read the dump file. Possible corruption of file doesn't exist."

**Comment by Naimarha** [comment]
Hi there, I could solve this issue using the "Update dump file" option in the "Tools" menu. Hope this works.

**Comment by yusiwen** [comment]
The site piratebay.org seems down currently, is there another way to get the dump file?

**Comment by DutchmanDavid** [comment]
A bit late, but you may be able to find an individual backup here: https://web.archive.org/web/*/https://thepiratebay.org/static/dump/csv/torrent_dump_full.csv.gz

**Comment by BachoSeven** [comment]
Yea, you can try this:

https://web.archive.org/web/20190924160713/https://thepiratebay.org/static/dump/csv/torrent_dump_full.csv.gz

**Comment by GreenLunar** [comment]
Hello,

Can someone please tell me how to decode the `HASH(B64)` column or refer me to a documentation?
```
$ grep slackware torrent_dump_full.csv.gz
2008-Dec-17 08:12:18;ZA/oTGE8F/ZjVR0hhommTorr6r4=;"slackware-12.2-dvd-iso";4154949894
```
I've tried to decode with `$ echo ZA/oTGE8F/ZjVR0hhommTorr6r4= | base64 -d` which didn't give a reasonable result.

TIA

**Comment by BachoSeven** [comment]
@GreenLunar Just guessing, but judging from the column's label, `HASH(B64)`, I would guess that it indeed is not just a BASE64-encoded string, but rather a string which was first encoded with b64, and then also passed through an hash function, which probably isn't (supposed to be) decodable... it's just an info hash to look for/identify torrents of which you know the hash.

**Comment by GreenLunar** [comment]
Let's take `slackware-12.2-dvd-iso`
```
$ grep "slackware" torrent_dump_full.csv 
2008-Dec-17 08:12:18;ZA/oTGE8F/ZjVR0hhommTorr6r4=;"slackware-12.2-dvd-iso";4154949894
```
Following size (`4154949894`) and date (`2008-Dec-17 08:12:18`), I guess this is the torrent https://thepiratebay.org/description.php?id=4582957

How do I go from `640FE84C613C17F663551D218689A64E8AEBEABE` to `ZA/oTGE8F/ZjVR0hhommTorr6r4=`?

**Comment by BachoSeven** [comment]
oh got it, to get the hash you just need to convert the raw decoded bytes to hexadecimal with `xxd`:

```
base64 -d <<< ZA/oTGE8F/ZjVR0hhommTorr6r4= | xxd -p
```

**Comment by GreenLunar** [comment]
Great! Thank you. How did you figure this out?

**Comment by BachoSeven** [comment]
The output was garbled bytes so I tried :)

Btw, if you place the hash in a base64 decoding website they automatically do this, but GNU `base64` doesn't.
